OpenRA.Mods.D2k/Traits/VariedCostMultiplier.cs

```diff
@@ -0,0 +1,39 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.D2k.Traits
+{
+	[Desc("Modifies the production cost of this actor for a specific queue or when a prerequisite is granted. " +
+		"Requires `VariedCostManager` on the world actor.")]
```

Comment:
While we wanted to make it so, the random value for each actor are shared for all players for balance reasons. That (as far as i could tell from the tests i did in the original while i coded these) wasn't the case in the original.

I have 2 ideas:
- Implement it so VairedCostManager can be defined on Player or World actor. Defining it on player actor would make it per player, and defining on the world.
- Add a boolean on the VariedCostManager trait, do the all the calculation still on there, but per player if the said boolean is true.

This is not really necessary here, and just for implementing the original behaviour, if someone wants to use. I just wanted to note.

Reply:
Cost changes between players is probably very bad for competitive play.